src/rust/engine/src/scheduler.rs

```diff
@@ -32,13 +32,19 @@ pub struct Session {
   preceding_graph_size: usize,
   // The set of roots that have been requested within this session.
   roots: Mutex<HashSet<Root>>,
+  // Flag used to determine whether to show engine execution progress.
+  display: Arc<Mutex<Option<EngineDisplay>>>,
```

Comment:
I don't think that the display needs sharing in other locations (yet?), so this probably doesn't need to be an Arc. You should also be able to move the `Option` outermost here: `Option<Mutex<EngineDisplay>>`

Reply:
Will remove the `Arc` for now, although I expect to need it soon. With regards to pulling the Option out, I promise we tried to make `Option<Arc<Mutex<_>>>` work, but there is a problem where doing:
```
let display: Option<EngineDisplay> = match maybe_display {
  Some(d) => Some(d.lock()) /* 1 */
  None => None
}
```

Made compilation fail, because a separate scope is created for `d`, and is dropped at point `/* 1 */`. Daniel tried to make it work for a while, and we decided that it was probably not worth it.

Reply:
would need to do:
```
match &maybe_display {
  &Some(ref d) => Some(d.lock()) /* 1 */
  &None => None
}
```
I think?